### PR TITLE
Simplified code for thread synchronization 2

### PR DESCRIPTION
**Pull request by emilwestergren**
@marcusnaslund peer review
Depends on https://github.com/magic-lang/ooc-kean/pull/1791

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Waiting until #1791 is merged.

**Comment by marcusnaslund** [comment]
Please rebase.

**Comment by emilwestergren** [comment]
Rebased

File: source/concurrent/Promise.ooc [hunk 1]
```diff
@@ -15,20 +15,23 @@ _PromiseState: enum {
 	Cancelled
 }
 
-Promise: abstract class {
+_Synchronizer: abstract class {
```
Comment (emilwestergren) [inline source/concurrent/Promise.ooc]: If you have a better name...

Comment (marcusnaslund) [inline source/concurrent/Promise.ooc]: Yeah, it bothered me that `Promise` and `Future` didn't have a base class in common. This is better.

Comment (marcusnaslund) [inline source/concurrent/Promise.ooc]: But I don't have a better name, I think this is OK.

**Comment by marcusnaslund** [comment]
Great job.

**Pull request merged**